code/game/machinery/chem_heater.dm

```diff
@@ -0,0 +1,159 @@
+	var/accept_drinking = FALSE
+	var/machine_strength = 0 //How much joules to add per process. Controlled by manipulators.
+	var/should_heat = TRUE
+	var/min_temperature = 50
+	var/max_temperature = 1000
+
+	component_types = list(
+		/obj/item/weapon/circuitboard/chem_heater,
+		/obj/item/weapon/stock_parts/manipulator,
+		/obj/item/weapon/stock_parts/manipulator,
+		/obj/item/weapon/stock_parts/manipulator
+	)
+
+/obj/machinery/chem_heater/attack_hand(mob/user)
+	user.set_machine(src)
+	interact(user)
+
+/obj/machinery/chem_heater/attackby(obj/item/O, mob/user)
+	if(default_deconstruction_screwdriver(user, O))
+		return
+	if(default_deconstruction_crowbar(user, O))
+		return
+	if(default_part_replacement(user, O))
+		return
+	if(istype(O, /obj/item/weapon/reagent_containers/glass) || istype(O, /obj/item/weapon/reagent_containers/food))
+		if(container)
+			user << "<span class='warning'>There is already \a [container] on \the [src]!</span>"
+			return
+
+		var/obj/item/weapon/reagent_containers/RC = O
+
+		if(!accept_drinking && istype(RC,/obj/item/weapon/reagent_containers/food))
+			user << "<span class='warning'>This machine only accepts beakers!</span>"
+			return
+
+		if(!RC.is_open_container())
+			user << "<span class='warning'>You don't see how \the [src] could heat up the reagents in \the [RC].</span>"
+			return
+
+		container =  RC
+		user.drop_from_inventory(RC,src)
+		user << "<span class='notice'>You set \the [RC] in \the [src].</span>"
+		updateUsrDialog()
+		return
+
+/obj/machinery/chem_heater/interact(mob/user as mob)
+	if(stat & BROKEN)
+		return
+	user.set_machine(src)
+	var/dat ="<html>"
+	dat += "<head><TITLE>Chem Heater MKI</TITLE><style>body{font-family:Garamond}</style></head>"
+	dat += "<body><H1>Chem Heater MKI</H1>"
+
+	dat += "<p>Power: <a href='?src=\ref[src];action=togglepower'>[use_power ? "On" : "Off"]</a></p>"
+	dat += "<p>Target Temp: [round(target_temperature)]K / [round(target_temperature - T0C,0.1)]C<br> "
+
+	dat += "("
+	dat += "<a href='?src=\ref[src];action=adjusttemp;power=-100'>----</a>|"
+	dat += "<a href='?src=\ref[src];action=adjusttemp;power=-25'>---</a>|"
+	dat += "<a href='?src=\ref[src];action=adjusttemp;power=-5'>--</a>|"
+	dat += "<a href='?src=\ref[src];action=adjusttemp;power=-1'>-</a>|"
+	dat += "<a href='?src=\ref[src];action=adjusttemp;power=1'>+</a>|"
+	dat += "<a href='?src=\ref[src];action=adjusttemp;power=5'>++</a>|"
+	dat += "<a href='?src=\ref[src];action=adjusttemp;power=25'>+++</a>|"
+	dat += "<a href='?src=\ref[src];action=adjusttemp;power=100'>++++</a>"
+	dat += ")</p>"
+
+	if(container)
+		dat += "<p>Beaker Temp: [round(container.get_temperature(),0.01)]K / [round(container.get_temperature() - T0C,0.1)]C (<a href='?src=\ref[src];action=removebeaker'>Remove</a>)</p>"
+	else
+		dat += "<p>No container loaded!</p>"
+
+	dat += "</body><html>"
+
+	user << browse(dat, "window=chem_heater")
+	onclose(user, "chem_heater")
+	return
+
+/obj/machinery/chem_heater/Topic(href, href_list)
```

Comment:
Psure you want a `. = ..(); if (.) return` here, no?

Reply:
solved